pachyderm/templates/pachd/deployment.yaml

```diff
@@ -20,10 +20,8 @@ spec:
   strategy: {}
   template:
     metadata:
-      {{- if .Values.pachd.storage.amazon.iamRole }}
       annotations:
-        iam.amazonaws.com/role: {{ .Values.pachd.storage.amazon.iamRole }}
-      {{- end }}
+        checksum/storage-secret: {{ include (print $.Template.BasePath "/pachd/storage-secret.yaml") . | sha256sum }}
```

Comment:
Ah, interesting.  pachd has to restart when you change the storage secret?

(If you mounted it as a volume and pachd could read the files, then you could update it without having to restart.)

Another common pattern here is naming the secret `pachyderm-storage-secret-{{that hash}}` and using the same name in `envFrom` below.  That way, if this update to the secret causes pachd to crash, you can very cleanly rollback.  Without doing that, you have to rely on gitops to handle the rollback, which is probably fine.

Reply:
My understanding was that this was a kube limitation? (ie, the env vars only got refreshed on container restart?) Pach actually accepts the storage secrets both ways (mount and env var) a follow on task there has been to consolidate on one.